### PR TITLE
Use subscription-manager instead of yum for listing enabled repos

### LINKED ISSUE #52: `yum repolist` in fact rhsm_repos leads to deadlock
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 3.8.7 (Agent), 4.10 (Server)
- Ruby: 2.0.0
- Distribution: RHEL 7
- Module version: 2.0.1-rc0

## How to reproduce (e.g Puppet code you use)

Add a package resource which is not yet installed.
This package should contain a `puppet apply`in its scripts, e.g.

```
%pre
puppet apply <<EOF
file { '/etc/motd':
  ensure => present,
}
EOF
```

You can lock yum by adding a pid to `/var/run/yum.pid`

```
echo $$ > /var/run/yum.pid
yum repolist
```

## What are you seeing

`puppet agent -t` hangs

## What behaviour did you expect instead

`puppet agent -t` finishes

## Output log

```
Another app is currently holding the yum lock; waiting for it to exit...
  The other application is: yum
...
```

## Any additional information you'd like to impart

`yum repolist` in https://github.com/voxpupuli/puppet-rhsm/blob/master/lib/facter/rhsm_repos.rb#L5 leads to a deadlock when another yum process should run during a `puppet agent -t` run.

### PR DESCRIPTION
**Pull request by kallies**
Fixes #52

### CODE REVIEW

File: lib/facter/rhsm_repos.rb [hunk 1]
```diff
@@ -1,8 +1,9 @@
+require 'facter'
```
Comment (bastelfreak) [inline lib/facter/rhsm_repos.rb]: this require shouldn't be needed, facts should work out of the box. Did you run into an error without it?

**Comment by kallies** [comment]
Good catch, this was a leftover from a local test. I removed it from the commit.

**Comment by alexjfisher** [comment]
This reverted the change made in https://github.com/voxpupuli/puppet-rhsm/pull/19 which was originally made because the fact gathering was so slow.
Is there anything that can be done to speed this up?  Maybe by implementing https://github.com/voxpupuli/puppet-rhsm/issues/52#issuecomment-331376919 ?

**Comment by kallies** [comment]
@alexjfisher the fact gathering is ~6s slower with `subscription-manager`. `yum` will break puppet runs. 

@speer had a discussion regarding this issue. There is a second issue:

If the Satellite server (or RHSM) is not accessible or sends http errorcodes.

1. Not accessible: Puppet runs are stale until `subscription-manager` times out (default timeout is 180s per http request and `subscription-manager repos --list-enabled` triggers ~7 requests)
2. Satellite/RHSM frontend answers with http errorcode: fact returns empty array -> if someone (like we do) evaluates this fact to enable repos, Puppet tries to enable all repos (again - not a big issue but def. not a nice behaviour). 

We were thinking about a caching mechanism which addresses these two issues.

Thinking of the comment you mentioned ".. or search for `enabled = 1` and all sections without enabled in `/etc/yum.repos.d/redhat.repo`": This should also address both issues.

**Pull request merged**